sky/adaptors/aws.py

```diff
@@ -98,21 +98,31 @@ def resource(service_name: str, **kwargs):
 
     Args:
         service_name: AWS resource name (e.g., 's3').
-        kwargs: Other options.
+        kwargs: Other options. We add max_attempts to the kwargs instead of
+            using botocore.config.Config() because the latter is not hashable.
     """
+    assert 'config' not in kwargs, (
+        'config should not be passed to resource(), '
+        'as botocore.config.Config() is not hashiable.')
```

Comment:
```suggestion
        'as botocore.config.Config() is not hashable.')
```

Reply:
I think it is hashable, but not in the way we expected (it inherits the hash method of default python objects)

```
>>> hash(botocore.config.Config())
269460405
>>> hash(botocore.config.Config())
269599567
>>> hash(botocore.config.Config())
269460312
```